### DISCUSSION TITLE
How to avoid sub-queries with 1-to-1 relations?

### DISCUSSION THREAD

**Discussion by leberknecht** [doctrine/doctrine2#9122] [Support Questions]
I have an simple 1-to-1 relation:
```php
class Institution {
    /**
     * @ORM\OneToOne(targetEntity="InstitutionSettings", mappedBy="institution")
     */
    private $institutionSettings;
}

class InstitutionSettings
{
    /**
     * @ORM\OneToOne(targetEntity="Institution", inversedBy="institutionSettings")
     **/
    private $institution;
```
When i fire this:
```php
//InstitutionRepository
$this->createQueryBuilder('i')
            ->join('i.segments', 's')
            ->where(':user MEMBER OF s.users')
            ->setParameter('user', $user)
            ->getQuery()->getResult();
```
Then the hydrator fires sub-queries to resolve the institution->institutionSettings relation. I tried to specify the relation as `lazy`, but it turns out that `UnitOfWork.php` has 
```php
// Inverse side of x-to-one can never be lazy
$class->reflFields[$field]->setValue($entity, $this->getEntityPersister($assoc['targetEntity'])->loadOneToOneEntity($assoc, $entity));
```
(which is a semi-nice behavior btw, options shouldnt be silently ignored..but ok)
For what im doing the related settings are not needed at all, so it would be totally fine if the relation is ignore or something, but i cant make that work. I also tried joining the settings: 
```php
$this->createQueryBuilder('i')
            ->leftJoin('i.institutionSettings', 'is')
            ->join('i.segments', 's')
            ->where(':user MEMBER OF s.users')
            ->setParameter('user', $user)
            ->getQuery()->getResult();
```
so now the data would be in the response from the db, but it doesnt help, the hydrater/UoW still creates sub-queries. 

Is there any way to not end up with sub-queries?

**Comment by jim-simpletire** [comment]
I have a similar issue. It would be great if the OneToOne association would respect the "fetch" value of LAZY or EXTRA_LAZY. However, if the OneToOne association is not going to respect the fetch values of LAZY or EXTRA_LAZY (even if only in certain situations like inverse side), then perhaps an exception being thrown in those circumstances would be appropriate when fetch is not explicitly EAGER on an inverse side of a OneToOne association. This would make the developer aware of exactly what could happen before going to production and allow them to fix it by explicitly setting fetch to EAGER and understand that doing so will incur the additional queries and potential performance issues.

**Comment by leberknecht** [comment]
Well, i guess thats two different problems..or maybe three:
a) how to avoid sub-queries
b) can associations be modified on the fly
c) fetch-options should not be silently ignored

i found:
a) when i switch the owning side, the `fetch="LAZY"` does actually work :) if someone picks that up: dont forget to modify the migration a bit, the diff that comes out of the migration tool will of course not migrate the IDs from one table to the other
b) turns out: they can :) 
```
$metaData = $this->_em->getClassMetadata(Institution::class);
unset($metaData->associationMappings['institutionSettings']);
```
actually did the trick (even without setting the modified data via `$this->_em->getMetadataFactory()->setMetadataFor(Institution::class, $metaData);` )
Note: obviously this is a very error-prone thing to do, as you wont even get a result proxy on the field this way.
c) Im pretty sure i read some threads about that topic already on the issue tracker at some point, but cant really find it now..another question would be why this also happens if the data is already joined.. hm.
